src/app/personside/dialogpanel/sendMelding/SendNyMelding.tsx

```diff
@@ -1,119 +1,172 @@
+import { isLoadedPerson } from '../../../../redux/restReducers/personinformasjon';
+import { capitalizeName } from '../../../../utils/stringFormatting';
+import AlertStripeInfo from 'nav-frontend-alertstriper/lib/info-alertstripe';
+import { getSaksbehandlerEnhet } from '../../../../utils/loggInfo/saksbehandlersEnhetInfo';
+import { NyMeldingValidator } from './validatorer';
+import TekstFelt from './TekstFelt';
+import VelgDialogType from './VelgDialogType';
+import { useRestResource } from '../../../../utils/customHooks';
+import { Undertittel } from 'nav-frontend-typografi';
+import Oppgaveliste from './Oppgaveliste';
+import { isPosting } from '../../../../rest/utils/postResource';
+
+export enum OppgavelisteValg {
+    MinListe = 'MinListe',
+    EnhetensListe = 'Enhetensliste'
+}
+
+export enum SendNyMeldingDialogTyper {
+    SamtaleReferatTelefon = Meldingstype.SAMTALEREFERAT_TELEFON,
+    SamtaleReferatOppmøte = Meldingstype.SAMTALEREFERAT_OPPMOTE,
+    SpørsmålSkriftlig = Meldingstype.SPORSMAL_SKRIFTLIG
+}
+
+export interface FormState {
+    tekst: string;
+    dialogType: SendNyMeldingDialogTyper;
+    tema?: Kodeverk;
+    sak?: JournalforingsSak;
+    oppgaveListe: OppgavelisteValg;
+    visFeilmeldinger: boolean;
+}
 
 const FormStyle = styled.form`
     display: flex;
+    margin-top: 1rem;
     flex-direction: column;
     align-items: stretch;
-    > * {
-        margin-bottom: 1rem;
+    .ReactCollapse--collapse .skjemaelement {
+        margin-bottom: 0;
     }
 `;
 
-const CollapseStyle = styled.div`
-    .skjemaelement {
-        margin: 0;
+const KnappWrapper = styled.div`
+    display: flex;
+    flex-wrap: wrap;
+    > * {
+        margin-top: 0.7rem;
+        margin-right: 0.5rem;
+        flex-grow: 1;
     }
 `;
 
 const tekstMaksLengde = 5000;
 
 function SendNyMelding() {
-    const initialDialogType = Meldingstype.SamtalereferatTelefon;
-    const [dialogType, setDialogType] = useState(initialDialogType);
-    const [tekst, setTekst] = useState('');
-    const [tekstFeil, setTekstFeil] = useState(false);
-    const [tema, setTema] = useState<Kodeverk | undefined>(undefined);
-    const [temaFeil, setTemaFeil] = useState(false);
-    const [visFeilMeldinger, setVisFeilmeldinger] = useState(false);
+    const initialState: FormState = {
+        tekst: '',
+        dialogType: SendNyMeldingDialogTyper.SamtaleReferatTelefon,
+        tema: undefined,
+        sak: undefined,
+        oppgaveListe: OppgavelisteValg.MinListe,
+        visFeilmeldinger: false
+    };
+    const [state, setState] = useState<FormState>(initialState);
+    const updateState = (change: Partial<FormState>) => setState({ ...state, visFeilmeldinger: false, ...change });
+    const personinformasjon = useSelector((state: AppState) => state.restResources.personinformasjon);
+    const enhet = getSaksbehandlerEnhet();
+    const postReferatResource = useRestResource(resources => resources.sendReferat);
```

Comment:
Trengs redux her? Og om det trengs, hvorfor trengs to forskjellige ressurser? :/

Reply:
poster på to forskjellige endepunkter avhengig av om det er spørsmål eller referat. Leser statusen i Dialogpanel for å vise kvitteringer.